src/pages/LogOutOldUserPage.js

```diff
@@ -0,0 +1,74 @@
+import React, {Component} from 'react';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import * as Session from '../libs/actions/Session';
+import FullScreenLoadingIndicator from '../components/FullscreenLoadingIndicator';
+import Log from '../libs/Log';
+import Navigation from '../libs/Navigation/Navigation';
+
+const propTypes = {
+    /** The parameters needed to authenticate with a short lived token are in the URL */
+    route: PropTypes.shape({
+        /** The name of the route */
+        name: PropTypes.string,
+
+        /** Unique key associated with the route */
+        key: PropTypes.string,
+
+        /** Each parameter passed via the URL */
+        params: PropTypes.shape({
+            /** AccountID associated with the validation link */
+            accountID: PropTypes.string,
+
+            /** Short lived token */
+            shortLivedToken: PropTypes.string,
+
+            /** URL to exit to */
+            exitTo: PropTypes.string,
+        }),
+    }),
+
+    /** The data about the current session which will be set once the user is authenticated and we return to this component as an AuthScreen */
+    session: PropTypes.shape({
+        /** The authToken for the current session */
+        authToken: PropTypes.string,
+
+        /** The authToken for the current session */
+        email: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    route: {
+        params: {},
+    },
+    session: {},
+};
+
+class LogOutOldUserPage extends Component {
+    componentDidMount() {
+        const email = lodashGet(this.props.route.params, 'email', '');
+        if (this.props.session && this.props.session.email !== email) {
+            Log.info('[LogOutOldUserPage] Different user signed in - signing out');
+            Session.signOutAndRedirectToSignIn();
+        }
+
+        // Set isNavigationReady so that we can navigate in the AuthScreens
+        Navigation.setIsNavigationReady();
```

Comment:
This is strange. Why would a view be controlling when the navigation is ready?

Reply:
I agree that it's a bit weird. Please see [this review conversation](https://github.com/Expensify/App/pull/8855#discussion_r863133914) with Marc where we discussed the problem. Later [he recommended](https://github.com/Expensify/App/pull/8855#discussion_r863229830) that I set the navigation ready on this page and that works well so I went with it.

Reply:
OK, I don't think I am 100% following that, but let me lay out why I think it's a problem to put it here. 

If there is other code added to the app that is also relying on the navigation being ready, then now that code becomes very highly coupled to this specific page (view). This tight coupling could lead to further headaches when things need to be moved or refactored.

On one hand, it's not a problem now, so it would be easy to dismiss my concern, but when it does become a problem, it's going to be a REAL PROBLEM then. It's basically just making our future selves work harder.

I'd like to offer another solution which maybe can accomplish what you're trying to do here while avoiding the tight coupling of the two pages.

## Suggestion
Add a new page to Auth Screens that is called something like `NavigationReadyDetector`. It's sole purpose is to render nothing (`render()` would just return `null`), and call `Navigation.setIsNavigationReady();` from `componentDidMount()`. 

That way, the code that is managing the ready state of the navigation lib is very obvious and clear and it wouldn't be tightly coupled to any other pages so anything could rely on it's functionality without worrying about being coupled to another view.

What do you think?

## Ideally
In a perfect world, we would figure out what the exact reason is that having this in the `onReady()` event isn't working and we would fix that in order to allow it to work (because it sounds like we all agree that's the expected behavior).

Reply:
I think that's a great idea to create a NavigationReadyDetector component! You are totally right that it would be a mess if future pages tried to rely on this one to detect if the navigation was ready.

>In a perfect world, we would figure out what the exact reason is that having this in the onReady() event isn't working and we would fix that in order to allow it to work (because it sounds like we all agree that's the expected behavior).

That would be great. We would have to get it merged into react navigation. Maybe there's a good reason that the callback only runs once? [This comment](https://github.com/react-navigation/react-navigation/issues/9170#issuecomment-750289596) mentions that one solution is to conditionally render the navigation container instead of the navigators, but I'm not sure that would work for our app.

Reply:
How do I always render the `NavigationReadyDetector` in the AuthScreens? The linkingConfig maps a path to a navigation state so I don't think it makes sense to render that component as a page with a route, especially since we alway want it to be rendered.

I think I could just render below the navigation container. Does that sound good or do you have other ideas?https://github.com/Expensify/App/blob/607e48dfe372184a93af47fc8f628a366ab2eb3e/src/libs/Navigation/AppNavigator/AuthScreens.js#L351

Reply:
The above didn't work for me. I'm not quite sure why but I'm guessing because it's not a navigation screen.

Reply:
Hm, I'm not sure why it wouldn't work to move it outside the navigation container. Have you tried rendering it in `AppNavigator/index.js` instead (the same place where `<AuthScreens>` is rendered)?

Reply:
I have been working on the Onyx PR, but while I'm waiting for feedback over there I tried your suggestion and it worked well.
> rendering it in AppNavigator/index.js instead (the same place where \<AuthScreens\> is rendered)?